registry/storage/driver/storagedriver.go

```diff
@@ -106,11 +106,11 @@ type FileWriter interface {
 }
 
 // PathRegexp is the regular expression which each file path must match. A
-// file path is absolute, beginning with a path components and may contain
-// more of them separated by slashes, where each path component is
+// file path is absolute, beginning with a slash or a path components and may
+// contain more of them separated by slashes, where each path component is
 // restricted to alphanumeric characters or a period, underscore, colon or
 // hyphen.
-var PathRegexp = regexp.MustCompile(`^([A-Za-z0-9._:-]+(/[A-Za-z0-9._:-]+)*)+$`)
+var PathRegexp = regexp.MustCompile(`^([A-Za-z0-9._:-]*(/[A-Za-z0-9._:-]+)*)+$`)
```

Comment:
I'm not sure why you would ever want to use relative paths with a storage driver.

I don't understand that PR. Why would you remove it then add it back? Why don't you just change back to the original regex?